src/rdkafka_fetcher.c

```diff
@@ -863,18 +865,23 @@ int rd_kafka_broker_fetch_toppars(rd_kafka_broker_t *rkb, rd_ts_t now) {
                 if (rkt_last != rktp->rktp_rkt) {
                         if (rkt_last != NULL) {
                                 /* Update PartitionArrayCnt */
-                                rd_kafka_buf_update_i32(rkbuf,
-                                                        of_PartitionArrayCnt,
-                                                        PartitionArrayCnt);
+                                rd_kafka_buf_finalize_arraycnt(rkbuf, of_PartitionArrayCnt,
+                                        PartitionArrayCnt);
                         }
-
-                        /* Topic name */
-                        rd_kafka_buf_write_kstr(rkbuf,
-                                                rktp->rktp_rkt->rkt_topic);
+                        if (rd_kafka_buf_ApiVersion(rkbuf) > 12)
+                                /* Topic ID */
+                                rd_kafka_buf_write_uuid(rkbuf, &rktp->rktp_rkt->rkt_id);
+                        else
+                                /* Topic name */
+                                rd_kafka_buf_write_kstr(rkbuf,
+                                                        rktp->rktp_rkt->rkt_topic);
+                        
+                        /* Topic tags */
+                        rd_kafka_buf_write_tags(rkbuf);
```

Comment:
This must be removed as it's done earlier only when `rkt_last != NULL` and before writing the next topic id or topic name
```suggestion
```